### DISCUSSION TITLE
useIsFetching: A misleading name?

### DISCUSSION THREAD

**Discussion by SoyDiego** [TanStack/query#6291] [Ideas] [answered]
Hello, I'm new to the community and I'm using TanStack Query. I'd like to hear opinions about the name `useIsFetching`.  I'm not sure if someone has proposed or thought about this topic, but the name can be a bit confusing. When you read 'useIsFetching,' you'd expect it to return a boolean, not a number.

I know the documentation is very clear and explains everything perfectly, but wouldn't it be better to use a name like:

- useQueryCount
- useLoadingQueries
- useActiveQueries

And the documentation could look something like this (example with useQueryCount):

![image](https://github.com/TanStack/query/assets/19764334/c69269a3-206f-45ba-bb47-f9e7c178d93b)

I don't come here to destroy anything, I just write something that crossed my mind when I was using the library.

Thank you very much for your time

**Comment by vossmalte** [comment, marked as answer]
If it’s well documented I do not see an issue with the name. 

Also: 0 is falsey, other numbers truthy. So conversion to Boolean works fine.

**Reply by TkDodo** [reply]
yeah, that was the original idea. Even if you are not interested in the actual count, you can still do:

```
const isFetching = useIsFetching()

if (isFetching) { ... }
```

and it will work like you'd expect.

Also, we are 5 minutes past the last major release, I'm not thinking about breaking api changes right now 😆

**Reply by SoyDiego** [reply]
Thanks both for your opinion.
I appreaciate your time. I didn't expect a change or generate breaking change with this. Only I wanted to know opinions and now all is clearer for me.
